client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -32,12 +35,15 @@ export const validate = ( { primary_email, primary_sms, secondary_email, seconda
 	}
 };
 
-export const handleRequestResetOptions = ( { dispatch }, { userData } ) => (
+export const handleRequestResetOptions = ( { dispatch }, { userData }, next, transit = page ) => (
```

Comment:
Hmm right, that makes it tricky.

Maybe this would be more suited outside the data layer? As far as I understand it, the data layer is there to consolidate data needs from multiple parts of the app, ensuring that if one piece of data is needed in two places it's only requested once. But if requesting this specific piece of data has the side effect of redirecting the user, wouldn't that make this specific request only useful in this context?